hajji/hello.S

```diff
@@ -49,8 +48,15 @@ start:
          * we anchor it at $LOAD.  Note that we don't collide with
          * the code because the stack will always remain below
          * (i.e. less than) $LOAD and grows downwards from there.
+         * disable intterupts when setting up the stack. If an
+         * interrupt occurs between the two MOVs then the stack
+         * may point at the wrong memory location and the interrupt
+         * may crash the system
          */
+        cli
+        movw %ax, %ss
         movw $LOAD, %sp
+        sti
```

Comment:
See my other comment about performance and running with interrupts on. If you aren't using interrupts in an application then turn them off at the start. Be aware that putting CLI and STI around the stack loading code generally wouldn't be an issue because the processor is suppose to keep interrupts off while changing a segment register and re-enable them automatically after the following instruction. If running on an old bare metal system with an ancient 8088 processor it is possible you have a buggy chip that doesn't honor this.